### PR TITLE
Use ts for related articles

### PR DESCRIPTION
**Pull request by eessex**
Quick update to the `relatedArticles` component (used for series sub-articles) to use typescript and styled components. Also limits articles to those from the same channel as the parent article.

### CODE REVIEW

File: src/client/apps/edit/components/content/sections/related_articles/index.tsx [hunk 1]
```diff
@@ -146,7 +160,17 @@ export class RelatedArticles extends Component {
...
           onChange={this.onAddArticle}
         />
-      </div>
+      </Flex>
     )
   }
 }
+
+const RelatedArticlePreview = styled.div`
+  div[class^="ArticleCard__ImageContainer"] {
```
Comment (ashleyjelks) [inline src/client/apps/edit/components/content/sections/related_articles/index.tsx]: Is it always img content in this container? I think you're following BEM here for css naming conventions, in which case its not clear if ImageContainer correctly describes the Element

File: src/client/apps/edit/components/content/sections/related_articles/index.tsx [hunk 2]
```diff
@@ -146,7 +160,17 @@ export class RelatedArticles extends Component {
...
+    min-height: 300px;
```
Comment (ashleyjelks) [inline src/client/apps/edit/components/content/sections/related_articles/index.tsx]: Why add this min-height here? Were these CSS props (background and min-height) defined in a stylesheet under the `RelatedArticles__preview` class previously?

File: src/client/apps/edit/components/content/sections/related_articles/index.tsx (hunk 1 above)
Comment (eessex) [reply src/client/apps/edit/components/content/sections/related_articles/index.tsx]: Good point, would be smarter to export from reaction here and target `ImageContainer` -- I can do this as a follow up, for now it is able to target correctly.

File: src/client/apps/edit/components/content/sections/related_articles/index.tsx (hunk 2 above)
Comment (eessex) [reply src/client/apps/edit/components/content/sections/related_articles/index.tsx]: This is a placeholder box for when an image is not present -- the empty list shows an empty article that demo's what the related article would look like.

File: src/client/components/autocomplete2/index.tsx [hunk 3]
```diff
@@ -79,6 +80,12 @@ export class Autocomplete extends Component<
       remote: {
         url,
         filter: filter || returnItems,
```
Comment (ashleyjelks) [inline src/client/components/autocomplete2/index.tsx]: This is just an additional safeguard to ensure the header is being set with validated user tokens?

Comment (eessex) [reply src/client/components/autocomplete2/index.tsx]: Yeah, we need to manually set the access token for Bloodhound, before we were running un-authenticated requests. (the end game is showing a mix of published/unpublished content on the autocomplete, but that is not yet implemented here).

Comment (ashleyjelks) [reply src/client/components/autocomplete2/index.tsx]: Nice!

File: src/client/apps/edit/components/content/sections/related_articles/index.tsx [hunk 4]
```diff
@@ -41,9 +51,13 @@ export class RelatedArticles extends Component {
         .query({ query: RelatedArticleQuery(idsToFetch) })
         .end((err, res) => {
           if (err) {
-            console.error(err)
+            new Error(err)
+          }
+          const articles =
+            res && res.body && res.body.data && res.body.data.articles
```
Comment (xtina-starr) [inline src/client/apps/edit/components/content/sections/related_articles/index.tsx]: Do we use lodash in this repo? We could probably use [get](https://lodash.com/docs/4.17.14#get) here if so.

```js
const article = get(res, "res.body.data.articles")
```

But that's minor and not at all a blocker.

**Pull request merged**